### DISCUSSION TITLE
Enabling partial text matches in dashboard filters.

### DISCUSSION THREAD

**Discussion by dolfandringa** [apache/superset#25358] [Ideas]
I would like to use the dashboard filters to use a value for `LIKE` queries using jinja templating. Currently, if I add a filter for a column of type "Value" to a dashboard, I can only select a full value from the dropdown. It isn't possible to enter anything for partial text matches. I see three optionsd to address this:
 1. Using the same Value filter type, but add an boolean/checkbox option to allow partial matches. Right now, when you partially type a value, and hit enter, it disappears. It would be great if the value would be kept, even if it doesn't completely match an existing value. This partial value could then be retrieved in jinja templates, and be used for instance like this in the underlying query: `column LIKE '%{{filter_values("filter_name")}}%'`
 2. Have a dedicated "Text" filter type that is just a text box, but works the same way as above with respect to jinja templates and the query. It just is a different filter typoe instead of adding a checkbox to the current Value filter type. 
 3. Have dedicated `beginswith`, `endswith` and `contains` filter types that don't require jinja templating in the queries. This is probably the nicest feature, but I guess also the hardest one to implement. 

I would really like people's opinion about this, and would be happy to contribute in the form of a SIP and PR if there is consensus.

**Comment by ssergeef** [comment]
HI! Did you find how to resolve this issue?  Thanks)
There is a way to use this jinja query - your_data like ({{ "'%" + "".join (filter_values('your_data')) + "%'" }}) , but superset adds to the end of your query IN filter(.

**Reply by olivierp9** [reply]
This worked for me
```
        {% if (filter_values('your_data') is defined) and filter_values('your_data')|length > 0 %}
            AND {% for value in filter_values('your_data', remove_filter=True) %}
                {% if value %} your_data LIKE {{ "N'%" + value + "%'" }} {% endif %}{% if not loop.last %} OR {% endif %}
            {% endfor %}
        {% endif %}
```

**Reply by johncassil** [reply]
Stuggling to make this work though -- It seems like the filter brings back all of the possible values in the column (or 1000 at a time) and you cannot proceed with text that doesn't exist in the possible values list?  I need a user to be able to just input arbitrary text and click apply.

**Reply by olivierp9** [reply]
yeah in my case it works half of the time... If the columns don't have time to load your can manage to apply a partial filter, but if they load too quickly then you can't. It's pretty annoying...

**Reply by johncassil** [reply]
I see -- it doesn't work for me because it keeps adding the IN filter as well as this one, but I may have misconfigured it. 

I just need some way to filter my whole dashboard up front with a simple text box because my data is such that there are hundreds of thousands of unique things -- that I need to partial match on.  

I guess superset might not be the right tool :cry:

**Reply by nickcarnevale** [reply]
Has this been solved?

**Reply by sfirke** [reply]
No, this is still not in Superset. Here's another example of someone using Jinja as a workaround: https://github.com/apache/superset/discussions/33625#discussioncomment-14634247

There's some discussion in that issue implying that this feature hasn't been prioritized because admins with huge datasets view it as a performance risk. That said, Evan wrote in that thread:

> We're open to a proposal and/or PR that would give filters a configurable operator, where a filter creator can select "in" (default) or other LIKE/ILIKE options with options like %x, %x%, or x%. Then we can add performance warnings, and everyone wins.

I agree with this. I would be happy to help such a feature get merged if anyone wants to contribute it.

**Reply by richardfogaca** [reply]
There's a open PR for this: https://github.com/apache/superset/pull/38470
